monitoring/voltage-monitoring/src/main/java/com/farao_community/farao/monitoring/voltage_monitoring/RaoResultWithVoltageMonitoring.java

```diff
@@ -0,0 +1,78 @@
+package com.farao_community.farao.monitoring.voltage_monitoring;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.commons.Unit;
+import com.farao_community.farao.data.crac_api.Instant;
+import com.farao_community.farao.data.crac_api.RemedialAction;
+import com.farao_community.farao.data.crac_api.State;
+import com.farao_community.farao.data.crac_api.cnec.VoltageCnec;
+import com.farao_community.farao.data.crac_api.network_action.NetworkAction;
+import com.farao_community.farao.data.rao_result_api.ComputationStatus;
+import com.farao_community.farao.data.rao_result_api.RaoResult;
+import com.farao_community.farao.data.rao_result_api.AbstractRaoResultClone;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class RaoResultWithVoltageMonitoring extends AbstractRaoResultClone {
+
+    private final RaoResult raoResult;
+    private final VoltageMonitoringResult voltageMonitoringResult;
+
+    public RaoResultWithVoltageMonitoring(RaoResult raoResult, VoltageMonitoringResult voltageMonitoringResult) {
+        super(raoResult);
+        this.raoResult = raoResult;
+        this.voltageMonitoringResult = voltageMonitoringResult;
+    }
+
+    @Override
+    public ComputationStatus getComputationStatus() {
+        if (!voltageMonitoringResult.getStatus().equals(VoltageMonitoringResult.Status.UNKNOWN)) {
+            return ComputationStatus.DEFAULT;
+        } else {
+            return ComputationStatus.FAILURE;
+        }
+    }
+
+    @Override
+    public double getVoltage(Instant optimizationInstant, VoltageCnec voltageCnec, Unit unit) {
+        if (!unit.equals(Unit.KILOVOLT)) {
+            throw new FaraoException("Unexpected unit for voltage monitoring result :  " + unit);
+        }
+        if (!optimizationInstant.equals(Instant.CURATIVE)) {
+            throw new FaraoException("Unexpected optimization instant for voltage monitoring result : " + optimizationInstant);
+        }
+        double upperBound = voltageCnec.getUpperBound(unit).orElseThrow();
+        double lowerBound = voltageCnec.getLowerBound(unit).orElseThrow();
+        double maxVoltage = voltageMonitoringResult.getMaxVoltage(voltageCnec);
+        double minVoltage = voltageMonitoringResult.getMinVoltage(voltageCnec);
+        if (Math.abs(upperBound - maxVoltage) < Math.abs(minVoltage - lowerBound)) {
```

Comment:
No abs needed. Example : when comparing margin +6 to margin -10, you should retain -10
```suggestion
        if (upperBound - maxVoltage < minVoltage - lowerBound) {
```